handler/src/main/java/io/netty/handler/ssl/SslContext.java

```diff
@@ -673,6 +673,204 @@ static ApplicationProtocolConfig toApplicationProtocolConfig(Iterable<String> ne
         return apn;
     }
 
+    /**
+     * Creates a builder for new client-side {@link SslContext}.
+     */
+    public static Builder newBuilderForClient() {
+        return new Builder(true);
+    }
+
+    /**
+     * Creates a builder for new server-side {@link SslContext}.
+     *
+     * @param keyCertChainFile an X.509 certificate chain file in PEM format
+     * @param keyFile a PKCS#8 private key file in PEM format
+     */
+    public static Builder newBuilderForServer(File keyCertChainFile, File keyFile) {
+        return new Builder(false).keyManager(keyCertChainFile, keyFile);
+    }
+
+    /**
+     * Creates a builder for new server-side {@link SslContext}.
+     *
+     * @param keyCertChainFile an X.509 certificate chain file in PEM format
+     * @param keyFile a PKCS#8 private key file in PEM format
+     * @param keyPassword the password of the {@code keyFile}, or {@code null} if it's not
+     *     password-protected
+     */
+    public static Builder newBuilderForServer(File keyCertChainFile, File keyFile,
+            String keyPassword) {
+        return new Builder(false).keyManager(keyCertChainFile, keyFile, keyPassword);
+    }
+
+    /** Builder for configuring a new SslContext for creation. */
+    public static final class Builder {
```

Comment:
Do I then keep SslContext.newServerBuilder() or do I move that to the builder as SslContextBuilder.forServer()?

Keeping it on SslContext seems a bit weird, since that would means that neither top-level class can create itself; you must use SslContext to create SslContextBuilder and SslContextBuilder to create SslContext. That doesn't seem strange to me when Builder is a nested class.

Putting the constructing method on SslContextBuilder means that you have to do an additional import for something that most people don't care about initially. When deprecating the existing newClientContext(void) and newServerContext(File, File) it seemed useful that the new approach was only an additional .build() so that there would be little argument to keep any of the existing factory methods because they were "easier" or "more convenient".